src/client.rs

```diff
@@ -871,12 +871,6 @@ where
                 Err(e) => return (delegation.terminating(), Err(e)),
             };
 
-            let version = if self.tuf.root().consistent_snapshot() {
-                MetadataVersion::Hash(value.clone())
```

Comment:
I think we want this conditional, just returning `MetadataVersion::Number(role_meta.version())` when consistent snapshot is enabled.